src/Examples/UnitModels/Reactors/gibbs_reactor_testing.ipynb

```diff
@@ -0,0 +1,1112 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "# Flowsheet Gibbs Reactor Simulation and Optimization of Steam Methane Reforming\n",
+    "\n",
+    "\n",
+    "## Learning outcomes\n",
+    "\n",
+    "\n",
+    "- Call and implement the IDAES GibbsReactor unit model\n",
+    "- Construct a steady-state flowsheet using the IDAES unit model library\n",
+    "- Connecting unit models in a flowsheet using Arcs\n",
+    "- Fomulate and solve an optimization problem\n",
+    "    - Defining an objective function\n",
+    "    - Setting variable bounds\n",
+    "    - Adding additional constraints \n",
+    "\n",
+    "\n",
+    "## Problem Statement\n",
+    "\n",
+    "Following the previous example of [Steam Methane Reformation in an Equilibrium Reactor](http://localhost:8888/notebooks/GitHub/EquilibriumReactors/equilibrium_reactor_testing.ipynb) (NEED TO UPDATE THIS HYPERLINK), this example solves the flowsheet using a Gibbs Reactor instead. The steam methane reformation example is adapted from S.Z. Abbas, V. Dupont, T. Mahmud, Kinetics study and modelling of steam methane reforming process over a NiO/Al2O3 catalyst in an adiabatic packed bed reactor. Int. J. Hydrogen Energy, 42 (2017), pp. 2889-2903. Typically, the process follows the chemical equations below:\n",
+    "\n",
+    "**C</sub>H<sub>4</sub> + H<sub>2</sub>O</sub> → C</sub>O</sub> + 3H<sub>2</sub>**  \n",
+    "**C</sub>O</sub> + H<sub>2</sub>O</sub> → C</sub>O<sub>2</sub> + H<sub>2</sub>**\n",
+    "\n",
+    "However, the GibbsReactor unit model solves the equilibrium by minimizing Gibbs free energy. Conveniently, this eliminates the need for a reaction package although a thermophysical package is still required.\n",
+    "\n",
+    "The flowsheet that we will be using for this module is shown below with the stream conditions. As in the prior example, we will be processing syngas and steam feeds of fixed composition to produce hydrogen. The process consists of a mixer M101 for the two inlet streams, a compressor to prepressurize the feed to the reaction pressure, a heater H101 to preheat the feed to the reaction temperature, and a EquilibriumReactor unit R101. We will assume cubic equation of state vapor behavior via Peng-Robinson thermodynamics for this flowsheet. The properties required for this module are available in the same directory:\n",
+    "\n",
+    "- msr_pr.py\n",
+    "\n",
+    "The state variables chosen for the property package are **total molar flows of each stream, temperature of each stream and pressure of each stream, and mole fractions of each component in each stream**. The components considered are: **CH4, H2O, CO, CO2, and H2** and the process occurs in vapor phase only. Therefore, every stream has 1 flow variables, 5 mole fraction variables, 1 temperature and 1 pressure variable. \n",
+    "\n",
+    "![](msr_flowsheet.png)\n",
```

Comment:
This png is the same as the equilibrium reactor example, with the equilibrium expressions that were used there.